interface/billing/payment_master.inc.php

```diff
@@ -246,7 +246,7 @@ function generate_list_payment_category($tag_name, $list_id, $currvalue, $title,
                         </div>
                         <div class="col-3">
                             <label class="control-label" for="TdUnappliedAmount"><?php echo xlt('Undistributed'); ?>:</label>
-                            <div id="TdUnappliedAmount" class="form-control bg-danger"><?php echo ($UndistributedAmount*1==0)? attr("0.00") : attr(number_format($UndistributedAmount, 2, '.', ','));?></div>
+                            <div id="TdUnappliedAmount" class="form-control bg-danger text-light"><?php echo ($UndistributedAmount*1==0)? attr("0.00") : attr(number_format($UndistributedAmount, 2, '.', ','));?></div>
```

Comment:
This and the one below also should not be needed. If this is, then it means something bad is happening (ie. another non-bootstrap class is not playing by the rules). We need to seek and destroy the bad class or override here.

Reply:
@bradymiller BS approach is it is your problem to address impact of your setting on other aspects.  bg-* changes only background color.  They do not automatically change the text-* perhaps because it may not be relevant to that tag or may be someone may like yellow text on red background.

If you want to set a project-wide standard, create another custom class or we used a global jquery function.

Reply:
@mdsupport ,
Good point. In light of day, would it make sense to not set bg/text colors and instead use things like alert-danger which take care of both and avoids different themes from having this issue?
@tywrenn , any input on this?

Reply:
Alerts will be appropriate here.  Better will be to begin using [form validation error mechanism ](https://getbootstrap.com/docs/4.0/components/forms/#server-side)that can highlight user and password fields with an error message below.

Reply:
While we are here, then, lets come up with a good solution for the use of `bg-danger` and also incorporate into the other 3 places it is used in the codebase. Seems like options are:
1. Ensure it has a text-light with it (like is done here)
2. A better way

Reply:
@bradymiller text colors/backgrounds should not be used for any form-control inputs. If the intent is to use them for validation purposes as @mdsupport stated, we must use the built-in validation methods BS4 [has stated here](https://getbootstrap.com/docs/4.4/components/forms/#validation).

Reply:
If only we could instead send a little voice clip shouting "Knock it off user".:)

Reply:
Then after three tries we default to "Okay dumb???, clearly you have no business using this fine software!" then, kick them to login.
Okay, i'm done:)

Reply:
And might as well then show them the following gif at the login screen:
![tenor (2)](https://user-images.githubusercontent.com/278968/77484655-54287800-6de8-11ea-913d-f92fb9546f0e.gif)

Reply:
Nice follow up. I think i'm a bad influence on you doctor!